### PR TITLE
T31123 install msgs (Review Only)

### PR DESCRIPTION
**Pull request by bkarnow**
Extending the autoinstall flow to allow non-generic messages.
To do this I reworked `step_app_not_installed` - Now searches first for a 'group' install message under `__app_common_install_name__`, then a `NOTINSTALLED` string in the quest's strings, then finally falls back on the generic message.
The Blender quests have been migrated over to this new flow - let me know what you think, if this works I'll migrate all the other quests over and add them to this PR before submitting.

https://phabricator.endlessm.com/T31123

### CODE REVIEW

**Comment by bkarnow** [comment]
I think having this as a draft is better for "don't merge" PRs

File: data/quests_strings/noquest.csv [hunk 1]
```diff
@@ -27,42 +27,37 @@ NOQUEST_POSITIVE,👍,,,,
...
-NOQUEST_NOTINSTALLED,"Uh-oh! It looks like this App isn't installed!
-
-You need to download it from the App Center if you want to play.
-
-Click the icon named <tt><span insert_hyphens=""false"" foreground=""#287A8C"" background=""#FFFFFF"">App Center</span></tt> on the Desktop, then search for the program you want to install.",daemon,,,
-NOQUEST_NOCONNECTION,"Uh-oh! It looks like you don't have an active internet connection.
-
-We need an active internet connection to display the steps and materials for this activity.
+NOQUEST_NOTINSTALLED_GENERIC,Uh-oh! It looks like the App this quest needs isn't installed! Would you like me to help you install it?,daemon,,,
+NOQUEST_AUTOINSTALL,"OK, I'll try to start the App install, this shouldn't take too long!",daemon,,,
```
Comment (erikos) [inline data/quests_strings/noquest.csv:31]: It is a bit unclear what happens here. In EOS GNOME Software is started (on the detail page) but the dialogue suggests that the app might be installed automatically without further intervention. But the user still has to start the installation.

File: eosclubhouse/libquest.py [hunk 2]
```diff
@@ -1653,16 +1658,29 @@ def step_app_not_installed(self):
...
+        Failing that, if there is a quest-specific _NOTINSTALLED string available, it is used.
+        Finally, it falls back on the generic NOTINSTALLED string.
         '''
-        action = self.show_choices_message('NOQUEST_NOTINSTALLED',
+
+        if self.__app_common_install_name__ is not None:
+            install_msg = 'NOQUEST_' + self.__app_common_install_name__ + '_NOTINSTALLED'
+        else:
+            has_install_msg = self._get_message_info('{}_NOTINSTALLED'.format(self._qs_base_id))
+            if has_install_msg:
```
Comment (erikos) [inline eosclubhouse/libquest.py:1674]: Could be reduced to a one liner.  
`install_msg = 'NOTINSTALLED' if has_install_msg else 'NOQUEST_NOTINSTALLED_GENERIC'`

**Review by erikos — CHANGES_REQUESTED** [review]

File: eosclubhouse/libquest.py [hunk 3]
```diff
@@ -1653,16 +1658,29 @@ def step_app_not_installed(self):
...
+            else:
+                install_msg = 'NOQUEST_NOTINSTALLED_GENERIC'
+
+        action = self.show_choices_message(install_msg,
                                            ('NOQUEST_POSITIVE', None, True),
                                            ('NOQUEST_NEGATIVE', None, False)).wait()
         if action.future.result():
-            self.show_message('NOQUEST_DESCRIPTION')
+            self.show_message('NOQUEST_AUTOINSTALL')
             self.wait_for_app_install(self.app, confirm=True, repo=self.__app_repository__)
```
Comment (erikos) [inline eosclubhouse/libquest.py:1684]: We can automatically install the app here. Then we should pass `confirm=False`. That would match better the corresponding current dialogue.

One thing I noticed is that the installation process can not be cancelled in the app center once initiated from the quest.

File: data/quests_strings/noquest.csv [hunk 4]
```diff
@@ -27,42 +27,37 @@ NOQUEST_POSITIVE,👍,,,,
...
 NOQUEST_NAV_BAK,❮,,,,
-NOQUEST_NOTINSTALLED,"Uh-oh! It looks like this App isn't installed!
-
-You need to download it from the App Center if you want to play.
-
-Click the icon named <tt><span insert_hyphens=""false"" foreground=""#287A8C"" background=""#FFFFFF"">App Center</span></tt> on the Desktop, then search for the program you want to install.",daemon,,,
-NOQUEST_NOCONNECTION,"Uh-oh! It looks like you don't have an active internet connection.
-
-We need an active internet connection to display the steps and materials for this activity.
+NOQUEST_NOTINSTALLED_GENERIC,Uh-oh! It looks like the App this quest needs isn't installed! Would you like me to help you install it?,daemon,,,
```
Comment (erikos) [inline data/quests_strings/noquest.csv:30]: We need to phrase it more clearly. When we install the app automatically, this dialogue should state it clearly. For example: Uh-oh! It looks like the App this quest needs isn't installed! Would you like me to install it for you?

**Pull request merged**